src/join_mesh.py

```diff
@@ -1,140 +1,254 @@
-        mesh = mesh_io.Mesh.load(os.path.join(dirname, str(i)+".txt"))
-        accumulated.append(mesh)
-    return accumulated
+    
+    logging.info("Starting partition-wise mesh merge")
+    joined_mesh = vtk.vtkUnstructuredGrid()
+    joined_points = vtk.vtkPoints()
+    joined_data_arrays = None
+    joined_cells = vtk.vtkCellArray()
+    joined_cell_types = []
+    offset = 0
 
-
-def join_mesh_recovery(dirname, partitions, recoveryPath):
+    for i in range(partitions):
+        fname = prefix + "_" + str(i) + ".vtu"
+        reader = vtk.vtkXMLUnstructuredGridReader()
+        reader.SetFileName(fname)
+        reader.Update()
+        part_mesh = reader.GetOutput()
+
+        logging.debug("File {} contains {} points".format(fname,part_mesh.GetNumberOfPoints()))
+        for i in range(part_mesh.GetNumberOfPoints()):
+            joined_points.InsertNextPoint(part_mesh.GetPoint(i))
+        
+        part_point_data = part_mesh.GetPointData()
+        num_arrays = part_point_data.GetNumberOfArrays()
+        
+        if joined_data_arrays is None:
+            joined_data_arrays = []
+            for j in range(num_arrays):
+                newArr = vtk.vtkDoubleArray()
+                newArr.SetName(part_point_data.GetArrayName(j))
+                joined_data_arrays.append(newArr)
+        
+        
+        for j in range(num_arrays):
+            array_name = part_point_data.GetArrayName(j)
+            logging.debug("Merging from file {} dataname {}".format(fname,array_name))
+            array_data = part_point_data.GetArray(array_name)
+            for k in range(array_data.GetNumberOfTuples()):
+                joined_data_arrays[j].InsertNextTuple(array_data.GetTuple(k))
+
+        for i in range(part_mesh.GetNumberOfCells()):
+            cell = part_mesh.GetCell(i)
+            vtkCell = vtk.vtkGenericCell()
+            vtkCell.SetCellType(cell.GetCellType())
+            idList = vtk.vtkIdList()
+            for j in range(cell.GetNumberOfPoints()):
+                idList.InsertNextId(cell.GetPointId(j)+ offset)
+            vtkCell.SetPointIds(idList)
+            joined_cell_types.append(cell.GetCellType())
+            joined_cells.InsertNextCell(vtkCell)
+
+        offset += part_mesh.GetNumberOfPoints()
+
+    joined_mesh.SetCells(joined_cell_types,joined_cells)
+    joined_mesh.SetPoints(joined_points)
+    for data_array in joined_data_arrays:
+        joined_mesh.GetPointData().AddArray(data_array)
+            
+    return joined_mesh
+
+
+def join_mesh_recovery(prefix : str, partitions : int, recoveryPath : str):
     """
-    Parition merge with full recovery
+    Partition merge with full recovery
 
-    This uses the originally paritioned mesh as recovery information.
-    The result is in the same very order as the original mesh.
-    Original cells will be preserved, but will grouped partition-wise.
-    Cells between partitions will be appended.
+    This saves original mesh.
```

Comment:
I am not sure but `This recovers the original mesh.` seems a bit more correct. Since it is not preserved instead reconstructed using `GlobalIDs` and recovery data

Reply:
Alright :+1: